yarn-project/circuit-types/src/tx/tx.ts

```diff
@@ -40,13 +40,16 @@ export class Tx {
     public readonly enqueuedPublicFunctionCalls: PublicCallRequest[],
   ) {
     if (this.unencryptedLogs.functionLogs.length < this.encryptedLogs.functionLogs.length) {
+      // TODO(Miranda): This error was not throwing in some cases, as logs are nested objects which would show len 1 even if no logs existed
+      // Many tests produce enc logs and no unenc logs, so this error should have been throwing even in good cases
+      // Hence commented out for now
       // This check is present because each private function invocation creates encrypted FunctionL2Logs object and
       // both public and private function invocations create unencrypted FunctionL2Logs object. Hence "num unencrypted"
       // >= "num encrypted".
-      throw new Error(
-        `Number of function logs in unencrypted logs (${this.unencryptedLogs.functionLogs.length}) has to be equal
-        or larger than number function logs in encrypted logs (${this.encryptedLogs.functionLogs.length})`,
-      );
+      // throw new Error(
+      //   `Number of function logs in unencrypted logs (${this.unencryptedLogs.functionLogs.length}) has to be equal
+      //   or larger than number function logs in encrypted logs (${this.encryptedLogs.functionLogs.length})`,
+      // );
```

Comment:
I noticed this error was actually not throwing in cases where number of enc logs > number of unenc (this is the case for many tests), because `functionLogs` can be nested objects. It's not actually checking anything, so I commented it out. Should we keep this check, but force it to work?